app/models/link.rb

```diff
@@ -48,7 +49,34 @@ def self.build(params)
 
 private
 
-  def reset_time_and_status
-    self.update_columns(status: nil, link_last_checked: nil)
+  def find_link_url
```

Comment:
You could rename this to `link_with_matching_url`. I think that might be marginally clearer in terms of what the method returns. Same for `find_existing_link_url` and `find_existing_homepage_url`.
Also I think it's more standard to use nouns as Ruby method names when they return an object (Although I have no authoritative source to back that up :) )